ops/model.py

```diff
@@ -1270,6 +1270,26 @@ def list_files(self, path: str, *, pattern: str = None,
         """
         return self._pebble.list_files(path, pattern=pattern, itself=itself)
 
+    def exists(self, path: str) -> bool:
+        """Return true if the path exists on the container filesystem."""
+        try:
+            self._pebble.list_files(path, itself=True)
+        except pebble.APIError as err:
+            if err.code == 404:
+                return False
+            raise err
+        return True
+
+    def isdir(self, path: str) -> bool:
+        """Return true if a directory exists at the given path on the container filesystem."""
+        try:
+            f = self._pebble.list_files(path, itself=True)[0]
```

Comment:
Minor: the `[0]` gets kind of hidden away at the end. Maybe consider:

```
try:
    files = self._pebble.list_files(path, itself=True)
except ...:
    ...
return files[0].type == pebble.FileType.DIRECTORY
```

Reply:
done